hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/storage/DiskBalancerConfiguration.java

```diff
@@ -55,6 +71,88 @@ public final class DiskBalancerConfiguration {
       description = "The max parallel balance thread count.")
   private int parallelThread = 5;
 
+  @Config(key = "should.run.default",
+      defaultValue = "false",
+      type = ConfigType.BOOLEAN,
+      tags = { DATANODE, ConfigTag.DISKBALANCER},
+      description =
+          "If DiskBalancer fails to get information from diskbalancer.info, " +
+              "it will choose this value to decide if this service should be " +
+              "running."
+  )
+  private boolean diskBalancerShouldRun = false;
+
+  public boolean getDiskBalancerShouldRun() {
+    return diskBalancerShouldRun;
+  }
+
+  public void setDiskBalancerShouldRun(boolean shouldRun) {
+    this.diskBalancerShouldRun = shouldRun;
+  }
+
+  @Config(key = "service.interval",
+      defaultValue = "60s",
+      type = ConfigType.TIME,
+      tags = { DATANODE, ConfigTag.DISKBALANCER},
+      description = "Time interval of the Datanode DiskBalancer service. " +
+          "The Datanode will check the service periodically and update " +
+          "the config and running status for DiskBalancer service. " +
+          "Unit could be defined with postfix (ns,ms,s,m,h,d). "
+  )
+  private long diskBalancerInterval = Duration.ofSeconds(60).toMillis();
+
+  public Duration getDiskBalancerInterval() {
+    return Duration.ofMillis(diskBalancerInterval);
+  }
+
+  public void setDiskBalancerInterval(Duration duration) {
+    this.diskBalancerInterval = duration.toMillis();
+  }
+
+  @Config(key = "service.timeout",
+      defaultValue = "300s",
+      type = ConfigType.TIME,
+      tags = { DATANODE, ConfigTag.DISKBALANCER},
+      description = "Timeout for the Datanode DiskBalancer service. "
+          + "Unit could be defined with postfix (ns,ms,s,m,h,d). "
+  )
+  private long diskBalancerTimeout = Duration.ofSeconds(300).toMillis();
+
+  public Duration getDiskBalancerTimeout() {
+    return Duration.ofMillis(diskBalancerTimeout);
+  }
+
+  public void setDiskBalancerTimeout(Duration duration) {
+    this.diskBalancerTimeout = duration.toMillis();
+  }
+
+  @Config(key = "version",
+      defaultValue = "ONE",
+      type = ConfigType.STRING,
+      tags = { DATANODE, ConfigTag.DISKBALANCER},
+      description = "The version of the disk balancer service"
+  )
+  private String diskBalancerVersion = "ONE";
+
+  public String getDiskBalancerVersion() {
+    return diskBalancerVersion;
+  }
+
+  public void setDiskBalancerVersion(String version) {
+    this.diskBalancerVersion = version;
+  }
+
+  public DiskBalancerConfiguration() {
+  }
+
+  public DiskBalancerConfiguration(Optional<Double> threshold,
+      Optional<Long> bandwidthInMB,
+      Optional<Integer> parallelThread) {
```

Comment:
The DiskBalancerConfiguration constructor and state variables for configuration do not appear to include the `balancePolicy` configuration that is documented in the design doc.  Is it an oversight?  It would be great to include it in the basic framework PR and support balancePolicies, default DefaultBalancePolicy, can balance between different disk mediums.

> include the following configurations:
> threshold: threshold is used to calculate the volume data density to decide which datanode is empty and needs balance.
> bandwidth: a parameter to control the balance speed, so that the balance job won't affect the datanode's IO performance.
> parallelThreads: a parameter that determines how many balance jobs are running simultaneously.
> balancePolicy: the policy to choose which disk to balance, for example, only balance between SSD disks.
> 
>

Reply:
@neils-dev Thanks for the review.

I was preparing to add the balancePolicy part in another ticket after the basic framework.